### PR TITLE
Feature/fdb 330 lustre fs

### PR DESCRIPTION
**Pull request by danovaro**
identify Lustre FS and use the lustre API only on supported filesystems

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/fdb?pullRequest=56) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/fdb?pullRequest=56) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: danovaro<br/>:white_check_mark: Ozaq<br/>:white_check_mark: ChrisspyB<br/>:x: simondsmart<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/fdb?pullRequest=56) it.</sub>

File: src/fdb5/io/LustreSettings.cc [hunk 1]
```diff
@@ -35,19 +38,24 @@ bool fdb5LustreapiSupported() {
 #endif
 }
 
-int fdb5LustreapiFileCreate(const char* path, LustreStripe stripe) {
+int fdb5LustreapiFileCreate(eckit::PathName path, LustreStripe stripe) {
```
Comment (simondsmart) [inline src/fdb5/io/LustreSettings.cc:41]: I would expect this to take a path by const reference, not copy?

File: src/fdb5/io/LustreSettings.cc [hunk 2]
```diff
@@ -35,19 +38,24 @@ bool fdb5LustreapiSupported() {
...
 #if defined(fdb5_HAVE_LUSTRE)
 
-    static bool lustreapi_silence = false;
+    struct statfs buf;
 
-    if(not lustreapi_silence) {
-        fdb5_lustreapi_silence_msg();
-        lustreapi_silence = true;
-    }
+    statfs(path.dirName().localPath(), &buf);
```
Comment (simondsmart) [inline src/fdb5/io/LustreSettings.cc:47]: Is this where we want to do this (i.e. on every file creation), or do we want to do it at the root/directory level? I don't have strong opinions.

**Review by ChrisspyB — COMMENTED** [review]
looks good to me

**Pull request merged**